GWFish/priors_modules/priors.py

```diff
@@ -0,0 +1,120 @@
+    in the differential comoving volume: 
+    p(dL) propto dV/dz * (ddL/dz)^(-1) / (1 + z)
+    where dL is the luminosity distance, given by dL = c/H0 * (1 + z) * int_0^z dz'/E(z')
+
+    Parameters:
+    x (float or array-like): Luminosity distance value(s) in Mpc at which to evaluate the PDF.
+    a (float): The lower bound of the uniform distribution in redshift. Defaults to 0.0001.
+    b (float): The upper bound of the uniform distribution in redshift. Defaults to 5.
+    
+
+    Returns:
+    float or array-like: The PDF value(s) at the given x.
+
+    """
+    # find the redshifts corresponding to the input luminosity distance values
+    zz = np.linspace(0.0001, 5, 1000)
+    dd = cosmo.luminosity_distance(zz).value
+    z = np.interp(x, dd, zz)
+    # norm_factor (so that probability sum up to 1) neglected
+    # remember to divide const.c by 1000 to convert to km/s
+    return np.where(np.logical_and(z >= a, z <= b), 
+                cosmo.differential_comoving_volume(z).value * (x / (1 + z) + (const.c.value / 1000) * (1 + z) / (cosmo.H(0).value * cosmo.efunc(z)))**(-1) / (1 + z), 0)
+
+def uniform_in_comoving_volume_pdf(x, a = 0.0001, b = 5):
+    """
+    Calculate the probability density function (PDF) of a uniform distribution 
+    in the differential comoving volume: 
+    p(dL) propto dV/dz * (ddL/dz)^(-1)
+    where dL is the luminosity distance, given by dL = c/H0 * (1 + z) * int_0^z dz'/E(z')
+
+    Parameters:
+    x (float or array-like): Luminosity distance value(s) in Mpc at which to evaluate the PDF.
+    a (float): The lower bound of the uniform distribution in redshift. Defaults to 0.0001.
+    b (float): The upper bound of the uniform distribution in redshift. Defaults to 5.
+
+    Returns:
+    float or array-like: The PDF value(s) at the given x.
+
+    """
+    # find the redshifts corresponding to the input luminosity distance values
+    zz = np.linspace(0.0001, 5, 1000)
+    dd = cosmo.luminosity_distance(zz).value
+    z = np.interp(x, dd, zz)
+    # norm_factor (so that probability sum up to 1) neglected
+    # remember to divide const.c by 1000 to convert to km/s
+    return np.where(np.logical_and(z >= a, z <= b), 
+                cosmo.differential_comoving_volume(z).value * (x / (1 + z) + (const.c.value / 1000) * (1 + z) / (cosmo.H(0).value * cosmo.efunc(z)))**(-1), 0)
+
+
+def uniform_in_distance_squared_pdf(x, a = 10, b = 10000):
+    """
+    Calculate the probability density function (PDF) of a uniform distribution
+    in the distance squared: p(dL) propto dL^2
+
+    Parameters:
+    x (float or array-like): The value(s) at which to evaluate the PDF in Mpc.
+    a (float, optional): The lower bound of the uniform distribution. Defaults to 10 Mpc.
+    b (float, optional): The upper bound of the uniform distribution. Defaults to 10000 Mpc.
+
+    Returns:
+    float or array-like: The PDF value(s) at the given x.
+
+    """
+    # norm_factor (so that probability sum up to 1) neglected
```

Comment:
If it's neglected then this is not a PDF (as the function name would suggest)

If it can be neglected in this case, why is it included in others (e.g. the uniform one?)

Reply:
- The Botev paper is cited in the overleaf and there is also the link to the github repo
- I will write the function for the priors
- the norm factor is not included because it slows down the code as it needs to be computed depending on the distance range
- I will have a look at your last point

Reply:
Right, I checked again and I'd missed the citation, sorry! For some reason, Ctrl+F did not work. 

I get that the normalization is not necessary and slows down the code, my main point is that the docstrings and function names indicate these to be probability distributions, which they are not if they're not normalized; also, some of them are normalized (e.g. uniform), some are not (e.g. the $d_L^2$ one) and some are normalized sometimes (e.g. the sine pdf: it includes the 0.5 factor, but if `a` and `b` are not 0 and $\pi$ that's not correct).

Reply:
You're totally right. I will eliminate the normalization and rename the functions accordingly. People can add normalization when needed

Reply:
Thanks, Jacopo!